src/main.ts

```diff
@@ -39,37 +39,77 @@ let avgBarplot = new Barplot(
 $(window).on("resize", function () {
   avgBarplot.resize();
 });
-let data: DataPoint[];
-data = [
-  { name: "Victoria", value: "5", description: "Victoria" },
-  { name: "Vancouver", value: "10", description: "Vancouver" },
-  { name: "Edmonton", value: "20", description: "Edmonton" },
-  { name: "Calgary", value: "28", description: "Calgary" },
-  { name: "Saskatoon", value: "34", description: "Saskatoon" },
-  { name: "Regina", value: "38", description: "Regina" },
-  { name: "Winnipeg", value: "40", description: "Winnipeg" },
-  { name: "Windsor", value: "43", description: "Windsor" },
-  { name: "London", value: "54", description: "London" },
-  {
-    name: "Kitchener, Cambridge, Waterloo",
-    value: "55",
-    description: "Kitchener, Cambridge, Waterloo",
-  },
-  {
-    name: "St. Catharines, Niagara",
-    value: "60",
-    description: "St. Catharines, Niagara",
-  },
-  { name: "Hamilton", value: "70", description: "Hamilton" },
-  { name: "Toronto", value: "81", description: "Toronto" },
-  { name: "Montreal", value: "84", description: "Montreal" },
-  { name: "Sherbrooke", value: "84", description: "Sherbrooke" },
-  { name: "Quebec City", value: "94", description: "Quebec City" },
-  { name: "Halifax", value: "98", description: "Halifax" },
-  { name: "St. John's", value: "100", description: "St. John's" },
-];
-avgBarplot.plot(data, [0, 0], [100, 100]);
-avgBarplot.updatePlot(data);
+
+
+/*********************
+*** DRAW MEAN PLOT ***
+**********************/
+drawMeanPlot(avgBarplot);
+
+
+/**
+ * @brief   Renders the left-side plot with city averages, sorted descending.
+ * @param   avgBarplot : The barPlot to render.
+ * @returns N/A
+ */
+function drawMeanPlot(avgBarplot : Barplot) {
+  let countryData = Data.getSyncData();
+  let meanCountryData : DataPoint[] = [];
+  countryData.forEach((d) => addMeanCity(d));
+
+  function addMeanCity(cityData : DataModel) {
+    let citySummary: DataPoint = {
+      name:         cityData.name,
+      description:  cityData.name,
+      value:        (avgCityData(cityData.data) * 100).toString()
+    };
+    meanCountryData.push(citySummary);
+  }
+
+  meanCountryData.sort(compareDataPoint);
+
+  avgBarplot.plot(meanCountryData, [0, 0], [100, 100]);
+  avgBarplot.updatePlot(meanCountryData);
+}
+
+
+/**
+ * \brief   Averages the city data.
+ * @param   cityData : The array of DataPoint describing the city.
+ * @returns Returns the average of the city's DataPoint values.
+ */
+function avgCityData(cityData: DataPoint[]): number {
+  let sum = 0.0;
+  let numElements = 0;
+  cityData.forEach((d) => addDataPointElement(d.value));
+  function addDataPointElement(dataPointValue: string) {
+    sum += parseFloat(dataPointValue);
+    ++numElements;
+  }
+  let avg = sum / numElements;
+  return avg;
+}
```

Comment:
Can confirm from my hardcoded averages this morning, we're getting the exact same values, well done! Could we draft up some unit tests for this method? Let me know if you'd like to pair program it as well.

Reply:
Thanks! I will draft up some unit tests and run them by you for approval.